### DISCUSSION TITLE
how sqlalchemy manage connection issue on multiple request at same time

### DISCUSSION THREAD

**Discussion by dev-nasir-iqbal** [sqlalchemy/sqlalchemy#11456] [Usage Questions]
I am developing Flask application  utilizing SQLAlchemy as the ORM for MySQL database interaction. Despite configuring SQLAlchemy with a default pool size of five connections, I have observed delays in fulfilling requests from the second user onwards, particularly noticeable when multiple users make requests simultaneously.

Symptoms:

Requests are queued behind the first request when multiple users make requests simultaneously.
Performance degradation occurs under concurrent user requests, leading to increased response times.
Configuration:

SQLAlchemy is configured with a default pool size of five connections.
Observations:

Subsequent user requests are waiting for the completion of the first request, even though database connections are available in the pool.
Long-running requests seem to block other incoming requests until completion.
Expected Behavior:

Efficient handling of concurrent user requests, utilizing the configured database connection pool to serve multiple requests simultaneously without queuing.

here is my flask_sqlalchemy initialization code

```
from flask import Flask, render_template, request, redirect, url_for
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
# Configure the database URI. Replace 'sqlite:///test.db' with your database URI.
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///test.db'
app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False  # Disable modification tracking
db = SQLAlchemy(app)
```

any suggestion regarding it, Thanks

**Comment by CaselIT** [comment]
Hi,

Are you sure your webserver is configured to be able to serve multiple request at the same time?

**Comment by dev-nasir-iqbal** [comment]
Sadly i came to know that this is the synchronous behavior of the flask,
But i am having the same issue on aws ec2, i am running my server with
geventwebsocket using profile (due to using flask_socketio) , but the issue
remain same

**Reply by CaselIT** [reply]
It does not really matter were you run your flask application. Can you run requests that do not use the db in parallel with your setup?

For further assistance of flask you should likely look at the flask support forums

**Comment by dev-nasir-iqbal** [comment]
Thank you for your valuable response, i will look into it
